File: Haskell/parser_combinator/parse_simple_csv.hs
import Text.ParserCombinators.Parsec

csvFile = endBy line eol
line = sepBy cell (char ',')
cell = many (noneOf ",\n")
eol = char '\n'

main = do
	cs <- getContents
	let res = parse csvFile "" cs

	case res of
		Left err -> print err
		Right x -> putStrLn $ show x

File: fsharp/parser_combinator/parse_simple_csv.fs

open System
open FParsec.Primitives
open FParsec.CharParsers

let cell = manyChars (noneOf ",\n")
let line = sepBy cell (pchar ',')
let csvFile = sepEndBy line newline


let cs = Console.In.ReadToEnd()
let res = run csvFile cs

match res with
| Success (v, _, _) -> Console.WriteLine(v)
| Failure (msg, _, _) -> Console.WriteLine(msg)
